src/ccl_f1d.c

```diff
@@ -30,26 +32,124 @@ ccl_f1d_t *ccl_f1d_t_new(int n,double *x,double *y,double y0,double yf)
     return NULL;
   }
 
-  spl->x0=x[0];
-  spl->xf=x[n-1];
   spl->y0=y0;
   spl->yf=yf;
+  spl->x_ini=x[0];
+  spl->x_end=x[n-1];
+  spl->y_ini=y[0];
+  spl->y_end=y[n-1];
+  spl->extrap_lo_type=extrap_lo_type;
+  spl->extrap_hi_type=extrap_hi_type;
 
+  // Compute derivatives
+  // Low-end
+  if(spl->extrap_lo_type == ccl_f1d_extrap_const) {
+    spl->der_lo = 0;
+  } 
+  else if(spl->extrap_lo_type == ccl_f1d_extrap_linx_liny) {
+    spl->der_lo = (y[1]-y[0])/(x[1]-x[0]);
+  }
+  else if(spl->extrap_lo_type == ccl_f1d_extrap_linx_logy) {
+    spl->der_lo = log(y[1]/y[0])/(x[1]-x[0]);
+  }
+  else if(spl->extrap_lo_type == ccl_f1d_extrap_logx_liny) {
+    spl->der_lo = (y[1]-y[0])/log(x[1]/x[0]);
+  }
+  else if(spl->extrap_lo_type == ccl_f1d_extrap_logx_logy) {
+    spl->der_lo = log(y[1]/y[0])/log(x[1]/x[0]);
```

Comment:
Do we want to let divide by zero errors just go through here, or should we attempt to catch them?

Reply:
Caught